code/game/machinery/OpTable.dm

```diff
@@ -71,13 +88,17 @@
 	return
 
 /obj/machinery/optable/proc/check_victim()
-	if(locate(/mob/living/carbon/human, src.loc))
-		var/mob/living/carbon/human/M = locate(/mob/living/carbon/human, src.loc)
-		if(M.lying)
-			src.victim = M
-			icon_state = M.pulse ? "[modify_state]-active" : "[modify_state]-idle"
-			return 1
-	src.victim = null
+	if(!victim || !victim.lying || victim.loc != loc)
+		suppressing = FALSE
+		victim = null
+		var/mob/living/carbon/human/H = locate() in loc
+		if(istype(H))
+			if(H.lying)
+				icon_state = H.pulse ? "[modify_state]-active" : "[modify_state]-idle"
+				victim = H
+	if(victim)

```

Comment:
This if-check needs to end in a `return 1` for the code to not fuck up. Right now it'll set the state to idle regardless of whether or not there's a victim present. And this function will never return true. Both of which are bugs.

Reply:
solved

Reply:
actually solved now